sdk/turing/router/config/log_config.py

```diff
@@ -210,3 +210,17 @@ def to_open_api(self) -> OpenApiModel:
             serialization_format=self.serialization_format.value
         )
         return super().to_open_api()
+
+
+class RouterVersionLogConfig(LogConfig):
```

Comment:
Just curious, is there any reason why this implementation and `BigQueryLogConfig`, `KafkaLogConfig` took a different implementation without dataclasses?

Reply:
My initial concern was with the subclasses (of `LogConfig`) `BigQueryLogConfig`, `KafkaLogConfig` and `RouterVersionLogConfig`  having non-default values for certain arguments, which would create an issue because the default constructor generated for these classes would insert these arguments without default values _after_ those with default values, which would be invalid. 

However, it appears that it's not an issue if the constructors are defined manually, and if the additional attributes are intialised in them. I've thus added the dataclass decorator for these subclasses, as well as for other instances where I originally thought would've posed problems.